### PR TITLE
new es operator

### PR DESCRIPTION
**Pull request by ruguodangshi**
### 添加一个新的operator，使用自定的的es镜像，es版本为2.4.1
使用2.4.1版本es的原因是这个版本在公司线上环境使用的最多，目前还没有升级到新版本。
该operator有如下的特点：
- 使用自定义的镜像
- 使用localPv作为es的存储
- 固定master节点数为3
- 支持自定义data节点数，默认为1（取决于nodeNum这个参数，data节点数等于nodeNum-3）

### CODE REVIEW

File: repository/elasticsearch-v2.4.1/docs/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# elasticsearch-v2.4.1
+Elasticsearch is an open source distributed search engine ，it is Large-scale used in China Unicom.
+Since our production environment has always used the 2.x version of es, and a lot of parameter optimizations have been done on this version, we have built this es operator on the basis of version 2.4.1.
+The image used in the operator has been passed to my own dockerhub repository，tag is "ruguodangshi/elasticsearch-unicom:2.4.1".
```
Comment (zmalik) [inline repository/elasticsearch-v2.4.1/docs/README.md:4]: can you add a link to the `Dockerfile` in the documentation?

File: repository/elasticsearch-v2.4.1/operator/params.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    default: 0.1
+  - name: cpuLimits
+    description: "limits of  cpus for  every pod"
+    default: 0.5
+  - name: ES_HEAP_SIZE
+    description: "jvm of es process"
+    default: 512Mi
+  - name: storageClassName 
+    description: "storageClassName of volumeClaimTemplates"
+    default: "local-storage"
```
Comment (zmalik) [inline repository/elasticsearch-v2.4.1/operator/params.yaml:28]: `local-storage` can be risky for the users, unless a complete graceful strategy is defined.
Let's use the default storage class by default and still give users an option to specify the storage class name if they don't want the default storage class.

File: repository/elasticsearch-v2.4.1/operator/templates/statefulset.yaml [hunk 3]
```diff
@@ -0,0 +1,149 @@
+apiVersion: v1
+kind: Service
```
Comment (zmalik) [inline repository/elasticsearch-v2.4.1/operator/templates/statefulset.yaml:2]: let's move `Service` objects to a separate `service.yaml` in `templates`

File: repository/elasticsearch-v2.4.1/operator/templates/statefulset.yaml [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        - name: MASTER_SVC
+          value: "{{ .Params.appName }}"
+        # mounting storage persistent volume completely on the data dir
+        volumeMounts:
+        - name: data
+          mountPath: /data/
+        - mountPath: /etc/localtime
+          name: time
+      volumes:
+      - name: time       #跟上面的名称对应
```
Comment (zmalik) [inline repository/elasticsearch-v2.4.1/operator/templates/statefulset.yaml:135]: let's add the comments in English

**Review by zmalik — CHANGES_REQUESTED** [review]
Thanks @ruguodangshi ! 
PR looks good to start with, we are working on having a `CONTRIBUTING.md` to help new contributors to start with new operators. 
As for now, I have left a few comments for the PR.

**Comment by meichstedt** [comment]
@ruguodangshi is there any particular reason why you added an additional ES operator? What about improving the existing one [here](https://github.com/kudobuilder/operators/tree/master/repository/elastic)?

File: repository/elasticsearch-v2.4.1/docs/README.md (hunk 1 above)
Comment (ruguodangshi) [reply repository/elasticsearch-v2.4.1/docs/README.md:4]: thanks!
I will close this pr and submit a new one.

**Pull request closed without merging**